sklearn/inspection/_partial_dependence.py

```diff
@@ -82,8 +85,8 @@ def _grid_from_X(X, percentiles, grid_resolution):
     values = []
     for feature in range(X.shape[1]):
         uniques = np.unique(_safe_indexing(X, feature, axis=1))
-        if uniques.shape[0] < grid_resolution:
-            # feature has low resolution use unique vals
+        if is_categorical[feature] or uniques.shape[0] < grid_resolution:
+            # feature is categorical or has low resolution use unique vals
```

Comment:
this is the only place were we make use of the `is_categorical` information in `partial_dependence()` right?

If yes, let's just remove this parameter altogether from `partial_dependence()`. We can just tell users that the grid resolution should always be higher than the max number of categories, which in most cases should be verified. And this is something we can enforce and check in `plot_partial_dependence()`, before `partial_dependence()` is called. This shouldn't lead to any problem in 99.999% of the use-cases of that function I think.

Once we remove the parameter from `partial_depdence()`, we don't have to worry about API discrepancy anymore, we can just bikeshed on what the parameter should be for `plot_partial_dependence()` :)

Reply:
Yes, that's right. Thanks @NicolasHug